src/sentry/services/hybrid_cloud/rpc.py

```diff
@@ -421,7 +421,18 @@ def dispatch_to_local_service(
     service, method = _look_up_service_method(service_name, method_name)
     raw_arguments = service.deserialize_rpc_arguments(method_name, serial_arguments)
     result = method(**raw_arguments.__dict__)
-    return result.dict() if isinstance(result, RpcModel) else result
+
+    if isinstance(result, RpcModel):
+        return result.dict()
+    if isinstance(result, list):
+        output = []
+        for item in result:
+            if isinstance(item, RpcModel):
+                output.append(item.dict())
+            else:
+                output.append(item)
```

Comment:
Minor concern 2: We don't currently nest any lists and I don't suggest we start. But maybe it would be cleaner or safer to recursively apply the same check in case `item` is another list.

We don't have to do that now. I could throw together a quick follow-up PR if you want to merge this.